source/get-started/pojo-qs.txt

```diff
@@ -0,0 +1,410 @@
+   );
+
+.. note::
+ 
+   The registries are checked in order until one returns a codec for the
+   requested class. The ``DefaultCodecRegistry`` should be first in the
+   list, and the ``PojoCodecProvider`` should always be the last
+   ``CodecProvider`` since it can provide a codec for almost any class.
+
+Using the CodecRegistry
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The following list describes ways to set the ``pojoCodecRegistry`` for
+use:
+
+- Set it when instantiating a ``MongoClient`` object:
+
+.. code-block:: java
+
+   MongoClientSettings settings = MongoClientSettings.builder()
+       .codecRegistry(pojoCodecRegistry)
+       .build();
+   MongoClient mongoClient = MongoClients.create(settings);
+
+- Use an alternative ``CodecRegistry`` with a ``MongoDatabase``:
+
+.. code-block:: java
+
+   database = database.withCodecRegistry(pojoCodecRegistry);
+
+- Use an alternative ``CodecRegistry`` with a ``MongoCollection``:
+
+.. code-block:: java
+
+   collection = collection.withCodecRegistry(pojoCodecRegistry);
+
+Inserting a POJO into MongoDB
+-----------------------------
+
+The codec registry automatically tries to create a POJO ``Codec`` for
+unknown classes. This allows you to use POJOs out of the box without
+any extra configuration.
+
+Before you can insert a POJO into MongoDB, create a
+``MongoCollection`` instance configured with the POJO class:
+
+.. code-block:: java
+
+   MongoCollection<Person> collection = database.getCollection("people", Person.class);
+
+Insert a Person Instance
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To insert a ``Person`` into the collection, use the collection's
+``insertOne()`` method:
+
+.. code-block:: java
+
+   Person ada = new Person("Ada Byron", 20, new Address("St James Square", "London", "W1"));
+   collection.insertOne(ada).subscribe(new OperationSubscriber<InsertOneResult>());
+
+Insert Multiple Person Instances
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To insert multiple ``Person`` instances, you can use the collection's
+``insertMany()`` method, which takes a list of ``Person`` instances.
+
+The following example will add multiple ``Person`` instances into the
+collection:
+
+.. code-block:: java
+
+   List<Person> people = asList(
+       new Person("Charles Babbage", 45, new Address("5 Devonshire Street", "London", "W11")),
+       new Person("Alan Turing", 28, new Address("Bletchley Hall", "Bletchley Park", "MK12")),
+       new Person("Timothy Berners-Lee", 61, new Address("Colehill", "Wimborne", null))
+   );
+   collection.insertMany(people).subscribe(new OperationSubscriber<InsertManyResult>());
+
+Query the Collection
+--------------------
+
+To query the collection, you can use the ``find()`` method.
+
+The following example prints all the ``Person`` instances in the
+collection:
+
+.. code-block:: java
+
+   collection.find().subscribe(new PrintToStringSubscriber<>());
+
+The example prints the following result:
+
+.. code-block:: none
```

Comment:
S: unless there's a specific reason to keep the input and output separate, you can use the `io-code-block `directive instead